src/queues/consumer.js

```diff
@@ -0,0 +1,150 @@
+                            append = ` with error ${utils.errorToString(e)}`;
+
+                        this.logger.warn(`Distributed queue: delivery attempt #${deliveryAttempts} to queue "${queue}" ` +
+                                         `failed${append}`);
+
+                        if (consumeOptions.requeueToTail) {
+                            if (!consumeOptions.maxRetries || deliveryAttempts < consumeOptions.maxRetries) {
+                                message.properties.headers = _.omit(message.properties.headers, 'x-death');
+                                let properties = _.merge(message.properties, {
+                                    headers: { 'x-delivery-attempts': ++deliveryAttempts }
+                                });
+
+                                return this.publisher.publishTo(queue, message.content.toString(), properties)
+                                    .then(() => channel.ack(message))
+                                    .then(() => debug(`message requeued to queue "${queue}"`));
+                            }
+                        }
+
+                        return channel.nack(message, false, false)
+                    } catch (e) {
+                        this.logger.error('Distributed queue: Consume error handling failed: ' + e.stack)
+                    }
+                }
+            }, consumeOptions);
+
+            if (this.exchangeName)
+                this.logger.info(`Distributed queue: Consuming messages from exchange "${this.exchangeName}", ` +
+                                 `topic "${topic}", queue "${queue}"`);
+
+            else if (queue)
+                this.logger.info(`Distributed queue: Consuming messages from queue "${queue}"`);
+
+            if (debug.enabled)
+                debug(`Consuming messages from channel "${channel.getDescriptor()}"${topic ? ` on topic "${topic}"` : ''} with options: `, _.omit(consumeOptions, 'logger'));
+
+            return channel;
+        } catch (e) {
+            this.logger.error('Distributed queue: Consume failed: ' + e.stack);
+        }
+    }
+
+    _restartConsumers() {
+        if (!this.consumers.length)
+            return debug(`Not restarting consumers. No consumers in queue`);
+
+        debug(`Restarting consumers. Consumers in queue: ${this.consumers.length}`);
+
+        let consumers = _.clone(this.consumers);
+        this.consumers.length = 0;
+
+        return utils.promiseWhile(() => consumers.length, () => {
+            let consumer = consumers.pop();
+            return this.consume(consumer.handler, consumer.topic, consumer.options)
+        })
+    }
+
+    _validateConsumeChannel(queue) {
+        if (this.consumers.length) {
+            let sameQueueConsumer = this.consumers.find(x => x.queue === queue);
+            if (sameQueueConsumer)
+                throw new Error('Multiple consumers registered to the same queue. If you meant to add bindings, use the "channel.addTopics" method instead');
```

Comment:
If I understand the error correctly, I think this message is more readable.
```suggestion
                throw new Error(`A consumer is already configured for queue "${queue}". If you meant to add bindings, use the "channel.addTopics" method instead`);
```